salt/infra-certs/init.sls

```diff
@@ -0,0 +1,58 @@
+include:
+  - crypto
+
+{% set ip_addresses = ["IP: 127.0.0.1"] -%}
+{% for _, interface_addresses in grains['ip4_interfaces'].items() %}
+  {% for interface_address in interface_addresses %}
+    {% do ip_addresses.append("IP: " + interface_address) %}
+  {% endfor %}
+{% endfor %}
+
+{% for cert in ['velum', 'salt-api'] %}
+/etc/pki/{{ cert }}.key:
+  x509.private_key_managed:
+    - bits: 4096
+    - require:
+      - sls:  crypto
+      - file: /etc/pki
+  file.managed:
+    - replace: false
+    - user: root
+    - group: root
+    - mode: 644
+
+/etc/pki/{{ cert }}.crt:
+  x509.certificate_managed:
+    - ca_server: {{ salt['mine.get']('roles:ca', 'ca.crt', expr_form='grain').keys()[0] }}
+    {% if cert == 'velum' %}
+    - signing_policy: external
+    {% else %}
+    - signing_policy: internal
+    {% endif %}
+    - public_key: /etc/pki/{{ cert }}.key
+    - CN: {{ grains['fqdn'] }}
+    - C: {{ pillar['certificate_information']['subject_properties']['C']|yaml_dquote }}
+    - Email: {{ pillar['certificate_information']['subject_properties']['Email']|yaml_dquote }}
+    - GN: {{ pillar['certificate_information']['subject_properties']['GN']|yaml_dquote }}
+    - L: {{ pillar['certificate_information']['subject_properties']['L']|yaml_dquote }}
+    - O: {{ pillar['certificate_information']['subject_properties']['O']|yaml_dquote }}
+    - OU: {{ pillar['certificate_information']['subject_properties']['OU']|yaml_dquote }}
+    - SN: {{ pillar['certificate_information']['subject_properties']['SN']|yaml_dquote }}
+    - ST: {{ pillar['certificate_information']['subject_properties']['ST']|yaml_dquote }}
+    - basicConstraints: "critical CA:false"
+    - keyUsage: nonRepudiation, digitalSignature, keyEncipherment
+    {% if ip_addresses|length > 0 %}
+    - subjectAltName: "{{ ", ".join(ip_addresses) }}"
+    {% endif %}
+    - days_valid: {{ pillar['certificate_information']['days_valid']['certificate'] }}
+    - days_remaining: {{ pillar['certificate_information']['days_remaining']['certificate'] }}
+    - backup: True
+    - require:
+      - sls:  crypto
+      - file: /etc/pki
```

Comment:
Maybe not the right time (taking into account or tight schedule) but I was wondering if it could de possible to refactor these `x509.certificate_managed` into some common place (maybe with a `jinja2` `import`, a Salt `extend`...)